engines/sci/engine/kgraphics32.cpp

```diff
@@ -360,29 +360,112 @@ reg_t kScrollWindow(EngineState *s, int argc, reg_t *argv) {
 }
 
 reg_t kScrollWindowCreate(EngineState *s, int argc, reg_t *argv) {
-	debug("kScrollWindowCreate");
-	kStub(s, argc, argv);
-	return argv[0];
+	reg_t object = argv[0];
+
+	// Ignoring argv[1], which is the number of lines to allocate.
+
+	SegManager *segMan = s->_segMan;
+	int16 borderColor = readSelectorValue(segMan, object, SELECTOR(borderColor));
+	int16 mode = readSelectorValue(segMan, object, SELECTOR(mode));
+	GuiResourceId fontId = (GuiResourceId)readSelectorValue(segMan, object, SELECTOR(font));
+	int16 backColor = readSelectorValue(segMan, object, SELECTOR(back));
+	int16 foreColor = readSelectorValue(segMan, object, SELECTOR(fore));
+	reg_t plane = readSelector(segMan, object, SELECTOR(plane));
+
+	Common::Rect r;
+	r.left = readSelectorValue(segMan, object, SELECTOR(nsLeft));
+	r.top = readSelectorValue(segMan, object, SELECTOR(nsTop));
+	r.right = readSelectorValue(segMan, object, SELECTOR(nsRight)) + 1;
+	r.bottom = readSelectorValue(segMan, object, SELECTOR(nsBottom)) + 1;
+	Common::Point p(r.left, r.top);
+
+	ScrollWindow *scrollWindow = new ScrollWindow(segMan, r, p, plane,
```

Comment:
It seems a little weird to me that the scroll window is created here and then passed to registerScrollWindow instead of just having a createScrollWindow or similar call on GfxControls32 instead, since this seems to be the only place that ScrollWindows are ever created. What is the thought in this regard? Is there some other hidden place that is also creating ScrollWindows?